Source/OrchBase/Composer.cpp

```diff
@@ -2373,6 +3072,20 @@ void Composer::writeThreadDevIDefs(ComposerGraphI_t* builderGraphI,
         devII << "&Thread_" << threadAddr << "_Context,";          // thread
         devII << "&Thread_" << threadAddr << "_DeviceTypes[0],";   // devType
         devII << devIdx << ",";                  // deviceID
+        
+        
+        // Retrieve the device index from the Supervisor map
+        devISuperIdxMap_t::iterator devISrch;
+        devISrch = builderGraphI->devISuperIdxMap.find(devI);
+        if (devISrch == builderGraphI->devISuperIdxMap.end())
+        {   // Something has gone wrong that we are going to ignore.
+            //TODO: Barf
```

Comment:
Flagging this TODO (is this a good candidate for a `#warning`?)

Reply:
Yeah, a warning and a fprint is probably in order. Will address